packages/api-fetch/src/index.js

```diff
@@ -78,9 +83,22 @@ const defaultFetchHandler = ( nextOptions ) => {
 	);
 
 	return responsePromise
-		.then( checkStatus )
-		.catch( ( response ) => parseAndThrowError( response, parse ) )
-		.then( ( response ) => parseResponseAndNormalizeError( response, parse ) );
+		// Return early if fetch errors. If fetch error, there is most likely no
+		// network connection. Unfortunately fetch just throws a TypeError and
+		// the message might depend on the browser.
+		.then(
+			( value ) =>
+				Promise.resolve( value )
+					.then( checkStatus )
+					.catch( ( response ) => parseAndThrowError( response, parse ) )
+					.then( ( response ) => parseResponseAndNormalizeError( response, parse ) ),
+			() => {
+				throw {
+					code: 'fetch_error',
+					message: __( 'You are probably offline.' ),
+				};
```

Comment:
This is a "parsed response", do you think we should avoid that and just return the "response" object if `parse === false`

Reply:
What do you mean with "parsed response"? This catches only raw `fetch` errors.

Reply:
when someone calls `apiFetch` with `parse:  false`  as an option, the expectation is that the promise will return a response object in both successful responses and failures.

Reply:
Oh, I see. So setting parse to false would result in the "raw" response? In this case it would be an `Error` instance (type error).

Reply:
mmm interesting, so there's no Response anyway in that case, maybe it's fine to leave your code then

Reply:
Right, the caught response is an `instanceof Error`. I'm fine with either direction. The most important thing for me is to provide a more helpful error message. The browser doesn't specify why `fetch` failed, but we can assume that it's the network connection if the request otherwise works fine.

Reply:
Another thing I could do it is replace the `Error` with a new one with a new message.